src/main/java/cubicchunks/worldgen/dependency/RegionDependency.java

```diff
@@ -36,40 +36,24 @@
 public class RegionDependency implements Dependency {
 	
 	private GeneratorStage targetStage;
-	
-	private Set<Long> requiredCubes;
-	
 	private Set<Requirement> requirements;
-
-	private int xLow;
-	private int xHigh;
-	private int yLow;
-	private int yHigh;
-	private int zLow;
-	private int zHigh;
 	
-	
-	public RegionDependency(GeneratorStage targetStage, int radius) {
+	public RegionDependency(Cube cube, GeneratorStage targetStage, int radius) {
```

Comment:
I would make it something like asking DependencyProvider for a dependency for each cube separately and let the DependencyProvider decide if it wants to use singleton or provide different dependency for each cube (and remove the Cube argument from constructor).